src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,136 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+
+    private MyEntry<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new MyEntry[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
+        if (key == null) {
+            putForNullKey(value);
+            return;
+        }
 
+        if (size + 1 > table.length * LOAD_FACTOR) {
+            resize();
+        }
+        int index = getIndex(key);
+        MyEntry<K, V> entry = new MyEntry<>(key, value);
+        if (table[index] == null) {
+            table[index] = entry;
+            size++;
+        } else {
+            MyEntry<K, V> current = table[index];
+            while (current.getNext() != null) {
+                if (Objects.equals(current.getKey(), key)) {
+                    current.setValue(value);
+                    return;
+                }
+                current = current.getNext();
+            }
+            if (Objects.equals(current.getKey(), key)) {
+                current.setValue(value);
+            } else {
+                current.setNext(entry);
+                size++;
+            }
+        }
+    }
+
+    private void putForNullKey(V value) {
```

Comment:
are you sure you need a separate method for this? try to combine with the main logic